mm2src/coins/eth/v2_activation.rs

```diff
@@ -322,37 +334,44 @@ pub async fn eth_coin_from_conf_and_request_v2(
 pub(crate) async fn build_address_and_priv_key_policy(
     conf: &Json,
     priv_key_policy: EthPrivKeyBuildPolicy,
+    path_to_address: &StandardHDCoinAddress,
 ) -> MmResult<(Address, EthPrivKeyPolicy), EthActivationV2Error> {
-    let raw_priv_key = match priv_key_policy {
-        EthPrivKeyBuildPolicy::IguanaPrivKey(iguana) => iguana,
+    match priv_key_policy {
+        EthPrivKeyBuildPolicy::IguanaPrivKey(iguana) => {
+            let key_pair = KeyPair::from_secret_slice(iguana.as_slice())
+                .map_to_mm(|e| EthActivationV2Error::InternalError(e.to_string()))?;
+            Ok((key_pair.address(), EthPrivKeyPolicy::Iguana(key_pair)))
+        },
         EthPrivKeyBuildPolicy::GlobalHDAccount(global_hd_ctx) => {
             // Consider storing `derivation_path` at `EthCoinImpl`.
-            let derivation_path: Option<StandardHDPathToCoin> = json::from_value(conf["derivation_path"].clone())
+            let derivation_path = json::from_value(conf["derivation_path"].clone())
```

Comment:
`conf` in this place is entire configuration and it's not possible to get a certain derivation_path without addressing by a coin in the "coins" array.

If I'm trying to get_my_address for ETH with the request

```json
{
    "userpass": "-",
    "method": "get_my_address",
    "params": {
        "coin": "ETH",
        "path_to_address": {"account": 1, "is_change": true, "address_index": 0},
        "protocol": "UTXO"
    },
    "mmrpc": "2.0"
}
```

I'm getting the answer: 

```
{"mmrpc":"2.0","error":"Get Eth address error: Error deserializing 'derivation_path': invalid type: null, expected a string","error_path":"lp_coins.eth.v2_activation","error_trace":"lp_coins:4025] eth:5434] v2_activation:349]","error_type":"GetEthAddressError","error_data":{"EthActivationV2Error":{"error_type":"ErrorDeserializingDerivationPath","error_data":"invalid type: null, expected a string"}},"id":null}
```

It's happening because conf structure does not comply to expectations

Reply:
> conf in this place is entire configuration and it's not possible to get a certain derivation_path without addressing by a coin in the "coins" array.

That's not true.

> If I'm trying to get_my_address for ETH with the request

It's only true to `get_my_address`/`get_eth_address` where the wrong `conf` was passed when this part was implemented. I can fix it though.

Reply:
Fixed `get_my_address` and added a test for it.